analysis/statistical-dataset-comparison.py

```diff
@@ -0,0 +1,306 @@
+import pandas as pd
+import spacy
+import matplotlib.pyplot as plt
+import textstat
+from collections import Counter
+import numpy as np
+import pandas as pd
+import yaml
+from nltk.corpus import stopwords
+import nltk
+
+# Load NLTK resources
+nltk.download('punkt')
+nltk.download('stopwords')
+
+# Load spaCy model
+nlp = spacy.load("en_core_web_sm")
+
+# Load datasets
+human_df = pd.read_excel("data/gold_annotations_05112024_old.xlsx")
+# model_df = pd.read_csv("data/gpt_generations_dataset_v17_2_10k_yaml.csv")
+model_df = pd.read_csv("data/gpt_generations_dataset_v17_newPrompt_10k_yaml.csv")
+original_count = len(model_df)
+model_df = model_df[model_df['sentence'] != "UNREALISTIC COMBINATION"]
+filtered_count = len(model_df)
+print(f"Removed {original_count - filtered_count} rows with 'UNREALISTIC COMBINATION'")
+
+# Extract sentences
+human_sentences = human_df['sentence'].dropna().astype(str).tolist()
+model_sentences = model_df['sentence'].dropna().astype(str).tolist()
+
+# Extract and drop missing YAML
+human_YAML_raw = human_df['YAML'].dropna().to_dict()
+model_YAML_raw = model_df['query'].dropna().to_dict()
+
+both_yamls = [human_YAML_raw, model_YAML_raw]
+results = []
+
+# Process separately
+for id, curr_yaml in enumerate(both_yamls):
+    # Counters
+    entity_count = 0
+    property_count = 0
+    relation_count = 0
+    errors = []
+
+    for idx, yaml_str in curr_yaml.items():
+        try:
+            parsed = yaml.safe_load(yaml_str)
+            entities = parsed.get("entities", [])
+            relations = parsed.get("relations", [])
+
+            entity_count += len(entities)
+            relation_count += len(relations)
+
+            for ent in entities:
+                props = ent.get("properties", [])
+                property_count += len(props)
+        except Exception as e:
+            errors.append((idx, str(e)))
+
+    # Normalize
+    sample_count = len(curr_yaml)
+    results.append({
+        "Entities": entity_count / sample_count,
+        "Properties": property_count / sample_count,
+        "Relations": relation_count / sample_count,
+        "Parsing Errors": len(errors) / sample_count
```

Comment:
I am not sure if this is necessary, because we always have a valid yaml as output that is derived from pydantic baseclass.